test/tracer_factory_test.cpp

```diff
@@ -3,12 +3,82 @@
 #include <datadog/tags.h>
 
 #include <catch2/catch.hpp>
+#include <cstdio>
+#include <stdexcept>
 
 #include "../src/tracer.h"
 #include "mocks.h"
 
 using namespace datadog::opentracing;
 
+namespace {
+
+// `EnvGuard` sets an environment variable to a specified value for the scope
+// of the `EnvGuard`, restoring the environment variables previous value
+// afterward.
+class EnvGuard {
+  std::string name_;
+  bool had_value_;
+  std::string old_value_;
+
+ public:
+  explicit EnvGuard(std::string name, const char *value) : name_(name) {
+    if (const char *const old_value = std::getenv(name.c_str())) {
+      had_value_ = true;
+      old_value_ = old_value;
+    } else {
+      had_value_ = false;
+    }
+
+    const bool overwrite = true;
+    ::setenv(name.c_str(), value, overwrite);
+  }
+
+  ~EnvGuard() {
+    if (had_value_) {
+      const bool overwrite = true;
+      ::setenv(name_.c_str(), old_value_.c_str(), overwrite);
+    } else {
+      ::unsetenv(name_.c_str());
+    }
+  }
+};
+
+// `TmpFile` creates a temporary file having the specified content, and deletes
+// the file at the end of the `TmpFile` scope.  A path to the temporary file is
+// accesible via the `name` member function.
+class TmpFile {
+  std::FILE *file_;
+  std::string name_;
+
+ public:
+  explicit TmpFile(ot::string_view content) {
+    char name_buffer[L_tmpnam];
+    if (std::tmpnam(name_buffer) == nullptr) {
```

Comment:
This emits a warning during the build; I don't know how to suppress it.

The alternative is to use a POSIX function outside of the C++ standard.  The advantage for keeping this in the language standard is that it won't cause issues for Windows builds.